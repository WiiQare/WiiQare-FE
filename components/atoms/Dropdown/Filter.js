import { HiChevronDown } from 'react-icons/hi';

const DropdownFilter = ({label, icon, items, onClick=() => {}}) => {
    return (
        <div className="dropdown dropdown-end w-full md:w-1/3 flex">
			<label tabIndex={0} className="bg-white border rounded-xl w-full py-3 px-4 flex justify-between items-center">
				<span className='flex gap-4 items-center'>
					{icon()}
					<span className='font-light'>{label}</span>
				</span>
				<HiChevronDown size={20} />
			</label>
			<ul tabIndex={0} className="w-full mt-14 dropdown-content menu p-2 bg-white border rounded-box text-gray-700 font-light">
				{items.map((item, index) => <li onClick={onClick} key={index}><a>{item}</a></li>)}
			</ul>
		</div>
    );
}

export default DropdownFilter;
