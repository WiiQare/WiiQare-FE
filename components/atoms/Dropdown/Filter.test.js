import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Filter from "./Filter";

describe("Filter", () => {
  const label = "Filter";
  const items = ["All", "T-Shirt", "Shirt", "Pants", "Shorts"];
  let component;
  beforeEach(() => {
    const res = render(<Filter label={label} items={items} />);
    component = res.container;
  });

  test("should render Filter component", () => {
    expect(component).toMatchSnapshot();
  });

  test("should render label", () => {
    expect(screen.getByText(label)).toBeInTheDocument();
  });

  test("should render items", () => {
    items.forEach((item) => {
      expect(screen.getByText(item)).toBeInTheDocument();
    });
  });
});
