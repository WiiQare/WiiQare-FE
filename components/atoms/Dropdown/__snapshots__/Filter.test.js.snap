// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render Filter component 1`] = `
<div>
  <div
    class="undefined dropdown dropdown-end flex"
  >
    <label
      class=" bg-white border rounded-xl w-full py-2 px-4 flex justify-between items-center undefined"
      tabindex="0"
    >
      <span
        class="flex mr-2 gap-4 items-center"
      >
        <span
          class="font-light"
        >
          Filter
        </span>
      </span>
      <svg
        fill="currentColor"
        height="20"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </label>
    <ul
      class="undefined w-full mt-14 dropdown-content menu p-2 bg-white border rounded-box text-gray-700 font-light"
      tabindex="0"
    >
      <li>
        <a>
          All
        </a>
      </li>
      <li>
        <a>
          T-Shirt
        </a>
      </li>
      <li>
        <a>
          Shirt
        </a>
      </li>
      <li>
        <a>
          Pants
        </a>
      </li>
      <li>
        <a>
          Shorts
        </a>
      </li>
    </ul>
  </div>
</div>
`;
